### PR TITLE
Add Int.fromStringWithRadix

### LINKED ISSUE #11: Int.fromStringWithRadix (parseInt) missing
The current binding has a fixed argument for the radix:
https://github.com/bloodyowl/rescript-js/blob/802ef39a804fb998c21c9f8159c0d20cf4385c3f/src/Js__Int.res#L19

### PR DESCRIPTION
**Pull request by Minnozz**
Fixes #11

### CODE REVIEW

File: src/Js__Int.res [hunk 1]
```diff
@@ -16,18 +16,17 @@ module Constants = {
...
-  let maybeInt = fromString(x)
+let fromStringWithRadix: (string, ~radix: int) => option<int> = (x, ~radix) => {
+  let maybeInt = Js__Float.parseIntWithRadix(x, ~radix)
   if Js__Float.isNaN(maybeInt) {
     None
   } else {
     Some(fromFloat(maybeInt))
   }
 }
+let fromString: string => option<int> = fromStringWithRadix(~radix=10)
```
Comment (DZakh) [inline src/Js__Int.res:30]: How about using `Number` to have it runtime free?

**Comment by DZakh** [comment]
I think the `Js` module is low-level enough so it makes sense to copy-pasting code instead of reusing it to reduce the number of function calls. What do you think?

**Comment by Minnozz** [comment]
@DZakh I'm not against copy-pasting here, but I wanted to avoid exporting a binding in `Js__Int.res` that is only for internal / wrapped use. I don't think that is possible without creating a `.resi` file currently?

Edit: To be clear, did you mean the call to `Js__Float.parseIntWithRadix` or the call to `fromStringWithRadix` in the same file?

**Comment by DZakh** [comment]
@Minnozz I also don't know how to avoid exporting a binding without a `.resi` file. There's `%%private`, but it only works with let bindings.

Talking about what I meant, I thought that maybe it's better to copy-paste `fromStringWithRadix` code for `fromString` instead of using it as an implementation detail:
```rescript
let fromStringWithRadix: (string, ~radix: int) => option<int> = (x, ~radix) => {
  let maybeInt = Js__Float.parseIntWithRadix(x, ~radix)
  if Js__Float.isNaN(maybeInt) {
    None
  } else {
    Some(fromFloat(maybeInt))
  }
}
let fromString: (string) => option<int> = (x) => {
  let maybeInt = Js__Float.parseIntWithRadix(x, ~radix=10)
  if Js__Float.isNaN(maybeInt) {
    None
  } else {
    Some(fromFloat(maybeInt))
  }
}
```
My idea is that `rescript-js` should put the efficiency of code execution over the size of the generated output. Even though it's just one function call.

**Comment by DZakh** [comment]
Also, I'm not sure whether this is intended behavior. Maybe it's better to return None in this case. @bloodyowl what do you think?
image

**Comment by bloodyowl** [comment]
Thanks for your PR! To address @DZakh's comment I'm closing this in favour of #16

**Pull request closed without merging**